thriftbp/client_pool.go

```diff
@@ -505,6 +523,11 @@ func newClientPool(
 
 		slug: cfg.ServiceSlug,
 	}
+
+	if cfg.ThriftHostnameHeader != "" {
```

Comment:
instead of conditionally add the middleware, it might be better to always add the middleware but the middleware itself conditionally add the header (don't add it if it's empty string)

Reply:
sure